### PR TITLE
Fix master check

### PR DESCRIPTION
**Pull request by mrlumbu**
A couple of scripts rely on checking if current branch is master. The function `gitBranch` was intended to do this, but does not behave as expected when branch is in detached state. In particular if one does:
```
git checkout <commit hash>
git rev-parse --abbrev-ref HEAD 
```
the output is `HEAD` rather than `master`, which is what the dependent functions expect.

This PR fixes that with a new function `getMasterHeadCommit` so that current commit can be compared with that.

### CODE REVIEW

File: vars/getMasterHeadCommit.groovy [hunk 1]
```diff
@@ -0,0 +1,6 @@
+String call() {
+    if (env.MASTER_HEAD_COMMIT == null) {
+        env.MASTER_HEAD_COMMIT =   sh(script: "git rev-parse origin/master", returnStdout: true).trim()
+    }
+    env.MASTER_HEAD_COMMIT
+}
```
Comment (bazbremner) [inline vars/getMasterHeadCommit.groovy]: Missing newline.

Comment (mrlumbu) [inline vars/getMasterHeadCommit.groovy]: fixed

**Comment by joshmyers** [comment]
Have you run this on build.ci with the failed build, pulling in this branch of of pay-jenkins-library?

**Comment by mrlumbu** [comment]
builds aren't failing, they're just not doing the right thing. I've checked it is now doing the right thing on a test branch here https://build.ci.pymnt.uk/job/pay-selfservice/job/test_jenkins/, where I printed out relevant vars to check they were what I expected (it wasn't a master build so couldn't directly check problem is solved, but very confident that this will fix it, and certainly won't make it worse)

**Comment by joshmyers** [comment]
So https://build.ci.pymnt.uk/job/pay-selfservice/job/test_jenkins/7/replay/ looks like it pulled in this branch and did what you expected.

**Pull request merged**